### DISCUSSION TITLE
Contract Modification Notice - More than one modification for the same contract

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#797] [Q&A] [answered]
When we try to send more than one contract modification for the same contract we get the following message:

**Message**:
`count(for $x in efac:Change/efac:ChangedSection/efbc:ChangedSectionIdentifier[fn:matches(normalize-space(./normalize-space(text())), '^CON-\d{4}$')], $y in /*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:ContractModification/efac:Change/efac:ChangedSection/efbc:ChangedSectionIdentifier[. = $x] return $y) = 1`
**Field:**
`For a contract modification, 'Section identifier' (BT-1501(s)-Contract)`
`ND-ContractModification_BT-1501_s_-Contract`

Can only one type of modification be submitted for each contract?
And only one modification per contract, even if the modification is of the same type?

Our example:

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/39ef91fa-ff67-4528-85a8-36cbf7089a68)

Thanks in advance,
Alex

**Comment by YvesJo** [comment, marked as answer]
Hi,
For a contract modification, only one efac:ContractModification is expected for a given contract.
It is however possible to specify multiple sections (efac:ChangedSection) that have been modified, amongst which one and only one should identify the contract.
The Contract amount is itself computed from the amounts of all the tenders (LotTenders) it is derived from.
Let's consider the contract CON-0001 is based on the following LotTenders TEN-0001, TEN-0002 and TEN-0003.
For TEN-0001 there is an amount increase (because of a quantity or unit price increase) then the amount at that level must be updated and that section also listed
For TEN-0002 there is an amount increase for whatever reason, then the amount at that level must be updated and that section listed
For TEN-0003 there was no modification and this should be kept unmodified and not listed as part of the modified sections.
When it comes to quantity of goods bought, then the Lot properties may need to be updated. For the exemple below, I'll suppose that this is the case for LOT-0002
For the reason, you'll have to select amongst the various reasons the one that represents the main change

```
<efac:ContractModification>
    <efbc:ChangedNoticeIdentifier>00368098-2023</efbc:ChangedNoticeIdentifier>
    <efac:Change>
        <efbc:ChangeDescription languageID="ENG">Put here a short description of the change</efbc:ReasonDescription>
        <efac:ChangedSection>
            <efbc:ChangedSectionIdentifier>CON-0001</efbc:ChangedSectionIdentifier>
        </efac:ChangedSection>
        <efac:ChangedSection>
            <efbc:ChangedSectionIdentifier>TEN-0001</efbc:ChangedSectionIdentifier>
        </efac:ChangedSection>
        <efac:ChangedSection>
            <efbc:ChangedSectionIdentifier>TEN-0002</efbc:ChangedSectionIdentifier>
        </efac:ChangedSection>
        <efac:ChangedSection>
            <efbc:ChangedSectionIdentifier>LOT-0002</efbc:ChangedSectionIdentifier>
        </efac:ChangedSection>
    </efac:Change>
    <efac:ChangeReason>
        <cbc:ReasonCode listName="modification-justification"> -- insert the code for the MAIN reason for the contract modification --</cbc:ReasonCode>
        <efbc:ReasonDescription languageID="ENG">Put here any relevant reason description</efbc:ReasonDescription>
    </efac:ChangeReason>
</efac:ContractModification>
```

**Reply by AlexAS81** [reply]
And if the same contract has two types of contract modifications, for example, mod-cir (Need for modifications because of circumstances which a diligent buyer could not predict) and mod-rev (Modifications based on review clauses or options). How can we reflect it in the physical model?

Thanks in advance,
Alex

**Reply by YvesJo** [reply]
The code for the main reason (i.e. the one having the greatest impact) should be used; extra information may be added to the main reason description which would then contain details for all the relevant reasons.
KR
